policies/design-process.md

```diff
@@ -0,0 +1,154 @@
+
+Design Steps
+------------
+
+The following design steps should be used to create a design document:
+
+1. (Optional) Identify key design sub-problems
+2. Identify candidate solutions
+3. Select solution
+4. Refine and document
+5. Agree solution
+
+### (Optional) Identify key design sub-problems
+
+Certain large problem domains may be split into smaller problem areas and a
+separate design document produced for each sub-problem. In such a case this
+policy should be applied recursively to address the design for each sub-problem
+individually before incorporating all of the outputs into a single large
+design document.
+
+For such large problems a design workshop would normally be held to identify
+the sub-problems that require individual designs.
+
+### Identify candidate solutions
+
+Once the problem area is understood, candidate solutions (one or more) that meet
+the requirements and constraints should be identified. Each candidate solution
+should be documented in only as much detail as is required to explain the
+primary characteristics of the solution approach so that it can be compared
+against the other candidate solutions and to determine how well it meets the
+requirements and constraints.
+
+In some cases a candidate solution may be identified that meets most of the
+requirements and constraints but not all. Where this happens the requirements
+and constraints that are not met by the candidate solution should be clearly
+stated.
+
+Typically the design team will be the people that identify and document the
+candidate solutions. In some cases it may be appropriate to hold a design
+workshop to collectively identify the candidate solutions. In this case it will
+typically still be necessary for the design team to document the solution
+options outside of the workshop.
+
+The design team may choose to make recommendations about the preferred solution
+option.
+
+### Select solution
+
+Once the candidate solutions have been identified and documented a solution
+must be selected by the OTC. There should be an opportunity for the OpenSSL
+community or non-OTC members to provide input into the decision. This could
+either be in the form of comments on a draft Pull Request describing the
+candidate solutions, or it could be in the form of an open invite for community
+members to join the workshop where the solution is selected.
+
+Optionally the OTC could select no solution and instead direct the design team
+to iterate over the candidate solutions again and either make refinements or
+identify additional options.
+
+### Refine and document
+
+In this step the design team provide further detail on the selected solution
+option. There should be sufficient detail to enable the implementation work to
+progress and to understand the trajectory of the solution into the future. Some
+details may be omitted until a later stage of the development project if
+appropriate and they are not needed until later. The assumptions of a solution
```

Comment:
maybe: `...if appropriate and such a level of detail is not needed until later development stages.`?